lrmd/tls_backend.c

```diff
@@ -260,6 +261,24 @@ lrmd_remote_listen(gpointer data)
         return TRUE;
     }
 
+#ifdef TCP_USER_TIMEOUT
+    {
+    int rc;
+    int optval;
+    long sbd_timeout = get_sbd_timeout();
+
+    if (sbd_timeout > 0) {
+        optval = sbd_timeout / 2; /* time to fail and retry before watchdog */
+        rc = setsockopt(csock, SOL_TCP, TCP_USER_TIMEOUT, &optval, sizeof(optval));
+        if (rc < 0) {
+            crm_perror(LOG_INFO, "writing TCP_USER_TIMEOUT (%d) on client socket failed", optval);
+            close(csock);
+            return TRUE;
+        }
+    }
+    }
+#endif
+
```

Comment:
I don't mind splitting out more functions. They should be in crm_internal.h, but they still need to be prefixed, because anything not static is exposed by the library, even when not part of our self-defined public API. The goal is to avoid linking conflicts with similarly-named user-defined functions.

Maybe crm_set_tcp_timeout() in utils.c, crm_get_sbd_timeout() in watchdog.c and crm_remote_tcp_init() in remote.c.